src/contact.rs

```diff
@@ -683,16 +676,12 @@ impl Contact {
                 .sql
                 .query_map(
                     "SELECT id FROM contacts
-                 WHERE addr!=?1
-                 AND id>?2
-                 AND origin>=?3
+                 WHERE addr NOT IN (SELECT addr FROM self_addrs)
```

Comment:
Maybe something like `WHERE NOT EXISTS (SELECT * FROM self_addrs WHERE addr=contacts.addr)` is faster. `EXISTS` is most commonly used for queries like this. `SELECT *` vs `SELECT 1` should be optimized properly, but for `NOT IN (SELECT )` i'm not sure.

Reply:
there is no table anymore./